diff --git a/plat/qemu/qemu_sbsa/sbsa_sip_svc.c b/plat/qemu/qemu_sbsa/sbsa_sip_svc.c
index 535f0eb..b37dcd0 100644
--- a/plat/qemu/qemu_sbsa/sbsa_sip_svc.c
+++ b/plat/qemu/qemu_sbsa/sbsa_sip_svc.c
@@ -30,6 +30,7 @@ static int platform_version_minor;
 #define SIP_SVC_GET_GIC_ITS SIP_FUNCTION_ID(101)
 #define SIP_SVC_GET_CPU_COUNT SIP_FUNCTION_ID(200)
 #define SIP_SVC_GET_CPU_NODE SIP_FUNCTION_ID(201)
+#define SIP_SVC_GET_CPU_TOPOLOGY SIP_FUNCTION_ID(202)
 #define SIP_SVC_GET_MEMORY_NODE_COUNT SIP_FUNCTION_ID(300)
 #define SIP_SVC_GET_MEMORY_NODE SIP_FUNCTION_ID(301)
 
@@ -53,6 +54,13 @@ static struct {
 	memory_data memory[PLAT_MAX_MEM_NODES];
 } dynamic_platform_info;
 
+static struct {
+	uint32_t sockets;
+	uint32_t clusters;
+	uint32_t cores;
+	uint32_t threads;
+} cpu_topology;
+
 void sbsa_set_gic_bases(const uintptr_t gicd_base, const uintptr_t gicr_base);
 uintptr_t sbsa_get_gicd(void);
 uintptr_t sbsa_get_gicr(void);
@@ -281,6 +289,30 @@ void read_platform_version(void *dtb)
 	}
 }
 
+
+void read_cpu_topology_from_dt(void *dtb)
+{
+	int node;
+	const fdt32_t *prop;
+
+	node = fdt_path_offset(dtb, "/cpus/topology");
+	if (node < 0) {
+		cpu_topology.sockets = 1;
+		cpu_topology.clusters = 1;
+		cpu_topology.cores = dynamic_platform_info.num_cpus;
+		cpu_topology.threads = 1;
+		return;
+	}
+	prop = fdt_getprop(dtb, node, "threads", NULL);
+	cpu_topology.threads = prop ? fdt32_ld(prop) : 1;
+	prop = fdt_getprop(dtb, node, "cores", NULL);
+	cpu_topology.cores = prop ? fdt32_ld(prop) : dynamic_platform_info.num_cpus;
+	prop = fdt_getprop(dtb, node, "clusters", NULL);
+	cpu_topology.clusters = prop ? fdt32_ld(prop) : 1;
+	prop = fdt_getprop(dtb, node, "sockets", NULL);
+	cpu_topology.sockets = prop ? fdt32_ld(prop) : 1;
+}
+
 void sip_svc_init(void)
 {
 	/* Read DeviceTree data before MMU is enabled */
@@ -305,6 +337,7 @@ void sip_svc_init(void)
 
 	read_platform_config_from_dt(dtb);
 	read_cpuinfo_from_dt(dtb);
+	read_cpu_topology_from_dt(dtb);
 	read_meminfo_from_dt(dtb);
 }
 
@@ -354,6 +387,13 @@ uintptr_t sbsa_sip_smc_handler(uint32_t smc_fid,
 			SMC_RET1(handle, SMC_ARCH_CALL_INVAL_PARAM);
 		}
 
+	case SIP_SVC_GET_CPU_TOPOLOGY:
+		SMC_RET5(handle, NULL,
+			cpu_topology.sockets,
+			cpu_topology.clusters,
+			cpu_topology.cores,
+			cpu_topology.threads);
+
 	case SIP_SVC_GET_MEMORY_NODE_COUNT:
 		SMC_RET2(handle, NULL, dynamic_platform_info.num_memnodes);
 
